Clarify process for merging changes after edits (#876)
dhall-lang/dhall-lang: .github/CONTRIBUTING.md

and `dhall-to-json`), but they do not count as extra votes because they are not
a separate reimplementation of the standard.

The rules for merging a change are:

* You can merge a change 7 days after the original submission, even without

>   positive votes, so long as there are not 50% or more negative votes

Reviewer: I think this would be slightly clearer again:

```suggestion
  positive votes, so long as there are less than 50% negative votes
```

If the change is approved quickly, there is still a minimum 3-day waiting
period before merging for changes to the standard.

The exact rules for merging a change are:

> * At least 7 days since the original submission if there are not more than

Reviewer: Is this clearer?

```suggestion
* At least 7 days since the original submission if there are at most
```

  50% votes against

> * At least 3 days since the original submission if you have 100% approval

Reviewer 2: What is 100%? Is it 100% of a given set of people, 100% of people tagged to review, or 100% of people who have participated in the review?

Author: 100% of votes according to the voting rules (i.e. 1 vote per implementation of the language)

Reviewer 3: That confused me too. Wouldn't "... if you have approval" be enough ?

Author: @Nadrieril: You only need 50% of votes for an approval.  100% of votes just expedites the process

Reviewer 3: My understanding was that you have to wait 3 days if you have approval (i.e. 50% votes), and having more than 50% votes doesn't change anything. Would that be a new rule that having 100% votes allows for faster merging ? I'm not sure what you're saying

Reviewer 3: Here's what I had in mind:

> The exact rules for merging a change are:
> 
> * At least 7 days since the original submission, unless there are 50% or more votes against
>
> * At least 3 days since the original submission if you have approval (i.e. 50% or more votes for)
>
> * At least 24 hours since the last change (not counting merging `master` into
  your branch)

Does that match what you had in mind ?

Author: @Nadrieril: That's close.  What I originally had in mind was that 3 days would require 100% of positive votes, but I could change it to 50%

Reviewer 3: Just to confirm: 3 days + 50% is the current behaviour, and you wanted to change it ? Or I've been reading it wrong all this time and the current behavior was 3 days + 100% ?

There's also that line "The absence of a vote counts as a vote in favor." that adds confusion: if after 3 days no one has voted either way, does that count as 100% votes ? If so, why the 7 days rule ?

Author: @Nadrieril: When I originally wrote it the intention was 3 days + 100% for the current behavior, but I haven't been enforcing it strictly

The 7 day mark is where the absence of a vote counts in favor.  However, I think a simpler way to word things is the way you did by just specifying the positive/negative vote thresholds at each boundary

Reviewer 3: Oh, I've been doing it wrong this whole time then x) I'm in favor of removing the notion of approval altogether and only mention those vote+time thresholds
